### ISSUE TITLE
No way for Reaper HUD to visualize abilities that will always reveal the Reaper

### ISSUE THREAD

**Issue by Iridar** [X2CommunityCore/X2WOTCCommunityHighlander#1315]
`UITacticalHUD::UpdateReaperHUD` will cap out the per-shot concealment loss at `class'X2AbilityTemplateManager'.default.SuperConcealShotMax`, which is 80% by default, when targeting non-objective targets.

Existing overrides don't provide enough functionality to bypass this logic.

Existing abilities that always break concealment, like Banish, just visualize as if they keep concealment, and rely on textual description to relay their mechanics.

**Comment by BlackDog86** [comment]
The banish ability is a bit strange, in that it doesn't use the normal rules for super-concealment loss. The Concealment rule set on the ability template is eConceal_Always, then it fires off an event which is listened to by SoulReaperListener, which manually triggers SourceUnit.BreakConcealment at the end of the ability.

The main 'trigger' if you like of the Reaper Shot HUD being displayed and used is related to the ```SelectedAbilityState.MayBreakConcealmentOnActivation(CurrentTargetID)``` check in UpdateReaperHUD but basically this function checks various aspects of the ability template for the concealment rules and returns true for stuff like 'shots with vektor rifle' and false for all of the non-hostile stuff or where it's explicitly defined on the template as alwaysconceal. Basically banish is short-circuiting that part of the HUD at the moment, so the correction for it should be to adjust the ability template, if we want it to display like a 100% chance of reveal on the shot HUD. In any case there are quite a few ways of fixing this so a bit of guidance on the direction to go in would be useful before I go too far into a rabbit hole!

**Comment by BlackDog86** [comment]
This is an example of the art of the possible:
https://github.com/user-attachments/assets/a616a850-2b2e-4e4b-9e1f-7d823192383f
I added a check for ```ActivatedAbilityState.GetMyTemplate().ConcealmentRule != eConceal_Never``` to both XCGS_Unit::OnAbilityActivated and UITacticalHUD::UpdateReaperHUD, along with setting SuperConcealedModifier to 100 if eConceal_Never is set on the ability template, then changed the banish concealment rule from eConceal_Always to eConceal_Never and commented out the bit of the eventlistener which breaks concealment after the ability fires. 

The visualisation isn't awful but I'd hesistate to say whether it's better than the base game. Still, a fix similar to this could be used to allow the Reaper HUD to still be used, rolled against etc. in a straightforward way - all that would be required would be setting eConceal_Never on the ability template and it should basically work. 

Happy to submit a draft PR but I still think the implementation needs a bit of work.

**Comment by Iridar** [comment]
Yeah Banish isn't the best example since it uses special logic to break concealment, but the problem is that there's basically nothing you can do to make the UI display 100% to reveal with currently available tools.

**Comment by Tedster59** [comment]
are we not able to add a hook to allow mods to override/update the reveal chance?

**Comment by Iridar** [comment]
We probably could, I'm just pointing out it hasn't been done yet.
